choreography/src/choreography/Fabrikam.Choreography.ChoreographyService/Services/EventRepository.cs

```diff
@@ -36,8 +37,8 @@ public async Task SendEventAsync(List<EventGridEvent> listEvents)
 
             try
             {             
-                var response = await eventGridClient.PublishEventsWithHttpMessagesAsync(eventGridHost, listEvents);
-                response.Response.EnsureSuccessStatusCode();
+                var response = await eventGridClient.SendEventsAsync(listEvents);
+                Assert.AreEqual(200, response.Status);
```

Comment:
Why are we using an `NUnit` assert in a non-test?   This is not something that we should be doing anywhere.    There's no reason to validate the status code here;  the pipeline will throw a [`RequestFailedException` ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/RequestFailedException.cs) should the send not be successful.

Reply:
The use of `EventException` below is incorrect;  there exception could very well be something related to the network or host and may not have any direct correlation with Event Grid or an event.   We should consider either extending logic to provide a more accurate context or using a type that is more general-purpose.

Reply:
It seems like was [`RequestFailedException`](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventgrid/Azure.Messaging.EventGrid#troubleshooting) at here.